src/TypesFinder/FindReturnType.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace BetterReflection\TypesFinder;
+
+use BetterReflection\Reflection\ReflectionMethod;
+use phpDocumentor\Reflection\Types\Context;
+use phpDocumentor\Reflection\Types\ContextFactory;
+use phpDocumentor\Reflection\DocBlock;
+use phpDocumentor\Reflection\Type;
+use BetterReflection\Reflection\ReflectionFunctionAbstract;
+
+class FindReturnType
+{
+    /**
+     * Given a function, attempt to find the return type.
+     *
+     * @param ReflectionFunctionAbstract $function
+     * @return Type[]
+     */
+    public function __invoke(ReflectionFunctionAbstract $function)
+    {
+        $context = $this->createContextForFunction($function);
+
+        $docBlock = new DocBlock(
+            $function->getDocComment(),
+            new DocBlock\Context(
+                $context->getNamespace(),
+                $context->getNamespaceAliases()
+            )
+        );
+
+        $returnTags = $docBlock->getTagsByName('return');
+
+        foreach ($returnTags as $returnTag) {
+            /* @var $returnTag \phpDocumentor\Reflection\DocBlock\Tag\ReturnTag */
+            return (new ResolveTypes())->__invoke($returnTag->getTypes(), $context);
+        }
+        return [];
```

Comment:
I don't think so, I don't think we should make any assumptions about return types. If they're there, we return them, if not, we don't. Plain and simple that way - if applications using Better Refletion want to make that assumption, they can do so simply.